roxie/roxiemem/roxiemem.cpp

```diff
@@ -5207,18 +5257,24 @@ void DataBufferBottom::addToFreeChain(DataBuffer * buffer)
 
 void DataBufferBottom::Release()
 {
-    if (atomic_dec_and_test(&count))
+    if (count.fetch_sub(1, std::memory_order_release) == 1)
+    {
+        //No acquire fence - released() is assumed not to access the data in this buffer
+        //If it does it should contain an acquire fence
         released();
+    }
 }
 
 void DataBufferBottom::released()
 {
     // Not safe to free here as owner may be in the middle of allocating from it
     // instead, give owner a hint that it's worth thinking about freeing this page next time it is safe
-    if (atomic_cas(&count, DEAD_PSEUDO_COUNT, 0))
+    unsigned expected = 0;
+    if (count.compare_exchange_strong(expected, DEAD_PSEUDO_COUNT, std::memory_order_release))
```

Comment:
Changed to release - although I suspect it would have got away with it, because the contents of the data buffer probably aren't used if this succeeds, so no need to ensure they are read consistently afterwards